### PR TITLE
DFC-13450 Replace Sendgrid with Gove Notify

### PR DESCRIPTION
**Pull request by Gurnaik-Randhawa**
- Replaced send of email with Gov Notify in place of SendGrid
- Added query input field to email that is sent out for call back.
- Added email entered by user for email sent out for call back.
- removed redundant email templates code and tests.

### CODE REVIEW

File: DFC.App.ContactUs.Services.EmailService/DFC.App.ContactUs.Services.EmailService.csproj [hunk 1]
```diff
@@ -17,6 +17,7 @@
       <IncludeAssets>runtime; build; native; contentfiles; analyzers; buildtransitive</IncludeAssets>
     </PackageReference>
     <PackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="3.1.9" />
+    <PackageReference Include="Notify" Version="2.9.0" />
     <PackageReference Include="SendGrid" Version="9.21.0" />
```
Comment (keenesolutions) [inline DFC.App.ContactUs.Services.EmailService/DFC.App.ContactUs.Services.EmailService.csproj:21]: Should we still have a reference to SendGrid now that we've moved to Notify?

File: DFC.App.ContactUs.Services.EmailService/NotifyEmailService.cs [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+        private readonly ILogger<NotifyEmailService<TEmailRequestModel>> logger;
+        private readonly INotifyClientProxy notifyClientProxy;
+        private readonly NotifyOptions notifyOptions;
+
+        public NotifyEmailService(
+            ILogger<NotifyEmailService<TEmailRequestModel>> logger,
+            INotifyClientProxy notifyClientProxy,
+            NotifyOptions notifyOptions)
+        {
+            this.logger = logger;
```
Comment (keenesolutions) [inline DFC.App.ContactUs.Services.EmailService/NotifyEmailService.cs:29]: Shouldn't you check for null on these constructor interface params?

**Review by IanCrisp — APPROVED** [review]

**Review by keenesolutions — COMMENTED** [review]
Just some basic comments. I'll leave the rest to Ian who has a better handle on the actual logic.

File: DFC.App.ContactUs.Services.EmailService/DFC.App.ContactUs.Services.EmailService.csproj (hunk 1 above)
Comment (Gurnaik-Randhawa) [reply DFC.App.ContactUs.Services.EmailService/DFC.App.ContactUs.Services.EmailService.csproj]: Removed

File: DFC.App.ContactUs.Services.EmailService/NotifyEmailService.cs (hunk 2 above)
Comment (Gurnaik-Randhawa) [reply DFC.App.ContactUs.Services.EmailService/NotifyEmailService.cs:28]: Added

**Review by keenesolutions — APPROVED** [review]
Looks good.

**Pull request merged**